pkg/ruler/compat.go

```diff
@@ -51,12 +47,14 @@ func (a *appendableAppender) AddFast(l labels.Labels, ref uint64, t int64, v flo
 }
 
 func (a *appendableAppender) Commit() error {
-	_, err := a.pusher.Push(a.ctx, client.ToWriteRequest(a.samples, client.RULE))
+	_, err := a.pusher.Push(a.ctx, client.ToWriteRequest(a.labels, a.samples, client.RULE))
+	a.labels = nil
```

Comment:
Curious why we're not doing `a.labels = a.labels[:0]`?

Reply:
This new line simply parallels the old `a.samples = nil` below.
I believe making them both do `[:0]`, as you suggest, is valid, and likely to be an improvement.